src/BrandEmbassyCodingStandard/Sniffs/Classes/ForbiddenStaticMethodCallSniff.php

```diff
@@ -0,0 +1,207 @@
+<?php declare(strict_types = 1);
+
+namespace BrandEmbassyCodingStandard\Sniffs\Classes;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use ReflectionClass;
+use ReflectionMethod;
+use SlevomatCodingStandard\Helpers\ClassHelper;
+use SlevomatCodingStandard\Helpers\StringHelper;
+use SlevomatCodingStandard\Helpers\TokenHelper;
+use function array_filter;
+use function array_map;
+use function assert;
+use function in_array;
+use function is_a;
+use function sprintf;
+use const T_CLASS;
+use const T_DOUBLE_COLON;
+use const T_SELF;
+use const T_STATIC;
+use const T_STRING;
+
+final class ForbiddenStaticMethodCallSniff implements Sniff
+{
+    private const FORBIDDEN_SELF_METHOD_CALL = 'ForbiddenStaticMethodCall';
```

Comment:
[o] nekonristentní název konstanty a její obsah, klidně ERROR_CODE (je to parametr `$code` nad `addFixableError`)

Reply:
obsah dle třídy, takže za mě `"ForbiddenSelfReferencingStaticMethodCalls"` ^^^